server/plugin.go

```diff
@@ -116,6 +126,29 @@ func (p *Plugin) OnActivate() error {
 	return nil
 }
 
+func (p *Plugin) MessageWillBePosted(c *plugin.Context, post *model.Post) (*model.Post, string) {
+	// If not enabled in config, ignore.
+	config := p.getConfiguration()
+	if !config.EnableCodePreview {
+		return nil, ""
+	}
+
+	msg := post.Message
+	info, err := p.getGitHubUserInfo(post.UserId)
+	if err != nil {
+		p.API.LogError("error in getting user info", "error", err)
+		return nil, ""
+	}
+	// TODO: apply the same optimization for /slash commands too.
+	p.initOnce.Do(func() {
+		p.githubClient = p.githubConnect(*info.Token)
+	})
```

Comment:
Agree with both points. However, I think doing this completely would require applying this logic for the /slash commands too.

It might be better to remove this optimization from this PR and then later do another PR which stores the `githubClient` at instance creation time, and reuse that for all calls. And also have an `updateClient` method which gets triggered on those user change events. That might be easier to review.